File: src/main/java/random/telegramhomebot/services/TimeLogConverter.java
package random.telegramhomebot.services;

import org.springframework.stereotype.Service;
import random.telegramhomebot.model.HostState;
import random.telegramhomebot.model.HostTimeLog;
import random.telegramhomebot.model.TimeLogDto;

@Service
public class TimeLogConverter {

    public TimeLogDto convertToDto(HostTimeLog timeLog) {
        return new TimeLogDto(getState(timeLog), timeLog.getCreatedDate());
    }

    private String getState(HostTimeLog timeLog) {
        return timeLog.getState() != HostState.FAILED
                ? HostState.REACHABLE.toString()
                : timeLog.getState().toString();
    }
}

File: src/main/java/random/telegramhomebot/services/TimeLogConverter.kt
package random.telegramhomebot.services

import org.springframework.stereotype.Service
import random.telegramhomebot.model.HostState
import random.telegramhomebot.model.HostTimeLog
import random.telegramhomebot.model.TimeLogDto

@Service
class TimeLogConverter {
    fun convertToDto(timeLog: HostTimeLog) = TimeLogDto(getState(timeLog), timeLog.createdDate)

    private fun getState(timeLog: HostTimeLog) =
        if (timeLog.state !== HostState.FAILED) HostState.REACHABLE.toString() else timeLog.state.toString()
}